02_knight_and_princess/matveev_alexey/task2.cpp

```diff
@@ -0,0 +1,79 @@
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+using VertexId = int;
+using EdgeId = int;
+
+struct Vertex {
+  const VertexId id;
+
+  explicit Vertex(const VertexId& _id) : id(_id) {}
+};
+
+struct Edge {
+  const EdgeId id;
+  const VertexId vertex_id1;
+  const VertexId vertex_id2;
+  Edge(const EdgeId& _id, VertexId _vertex_id1, VertexId _vertex_id2)
+      : id(_id), vertex_id1(_vertex_id1), vertex_id2(_vertex_id2) {}
+};
+
+class Graph {
+ public:
+  bool HasVertex(const VertexId& _vertex_id) {
```

Comment:
1. `Naming Convention`
2. Нет необходимости в `_`, достаточно просто `vertex_id` или `id`
3. Данная функция не изменяет родительский объект, а значит должна быть помечена `const`

```suggestion
  bool hasVertex(const VertexId& id) const {
```